File: 1212-sequential-digits/sequential-digits.java
class Solution {
    public List<Integer> sequentialDigits(int low, int high) {
        int[] q = new int[45];
        int n = 0;
        
        for (int i=1; i < 10; i++) {
            q[n++] = i;
        }

        for (int i=0; i<n; i++) {
            int d = q[i] % 10;
            if (d < 9) {
                q[n++] = q[i] * 10 + d + 1;
            }
        }

        List<Integer> res = new ArrayList<>();
        for (int x: q) {
            if (x >= low && x <= high) {
                res.add(x);
            }
        }
        
        return res;
    }
}

File: 1212-sequential-digits/sequential-digits.py
class Solution:
    def sequentialDigits(self, low: int, high: int) -> List[int]:
        q = list(range(1, 10))
        for x in q:
            d = x % 10
            if d < 9:
                q.append((x * 10) + (d + 1))
        return [x for x in q if low <= x <= high]